README.md

```diff
@@ -211,7 +211,71 @@ The metrics associated with this dataset will be printed in the terminal. These
 | Sequence Utilization | Average number of non-padding tokens in a sequence divided by sequence length. | The percent of the tokens in each sequence are actually used for training. This number imrpoved be changed by using different `input_packing_config`. The packing styles from highest sequence utilization to lowest are: `full`, `greedy::truncate_left` (or truncate_right), `greedy::drop`, `single::truncate_left` (or truncate_right), `single::drop`.|
 | Seq Completion Utilization | Average number of completions tokens in a sequence divided by sequence length. | The percent of the tokens in a sequence are learned.|
 
+## Validating training parameters with dataset metadata
 
+To help improve speed and cross-checking we now provide a metadata file along with the dataset. This file is located right under the `output_dir` as `metadata.yaml`. This file is used along with a custom pydantic model which you can import from this library which will verify the dataset parameters and the training parameters. This can be used as a way to catch bugs before training begins.
+#### Structure of Metadata file
+```
+max_seq_length: int
+token_type_ids: bool
+vocab_size: int
+tokenizer_model_type: str
+number_of_training_files: int
+number_of_dev_files: Optional[int]
+number_of_test_files: Optional[int]
```

Comment:
should not be necessary! But I guess left that in there for now in case need to use it for something in the future or if you want to see if there are test data files in the dataset!

Reply:
Ok makes sense, I general like to not add unused code until it is necessary, but this sounds good